src/d/actor/d_a_obj_hasu2.cpp

```diff
@@ -139,180 +73,182 @@ extern actor_process_profile_definition g_profile_Obj_MHasu = {
   fopAc_CULLBOX_CUSTOM_e,            // cullType
 };
 
-/* 80C18AFC-80C18B28 000054 002C+00 2/2 0/0 0/0 .data            __vt__Q29nObjMHasu12daObjMHasu_c */
-SECTION_DATA extern void* __vt__Q29nObjMHasu12daObjMHasu_c[11] = {
-    (void*)NULL /* RTTI */,
-    (void*)NULL,
-    (void*)CreateHeap__Q29nObjMHasu12daObjMHasu_cFv,
-    (void*)Create__Q29nObjMHasu12daObjMHasu_cFv,
-    (void*)Execute__Q29nObjMHasu12daObjMHasu_cFPPA3_A4_f,
-    (void*)Draw__Q29nObjMHasu12daObjMHasu_cFv,
-    (void*)Delete__Q29nObjMHasu12daObjMHasu_cFv,
-    (void*)IsDelete__16dBgS_MoveBgActorFv,
-    (void*)ToFore__16dBgS_MoveBgActorFv,
-    (void*)ToBack__16dBgS_MoveBgActorFv,
-    (void*)__dt__Q29nObjMHasu12daObjMHasu_cFv,
-};
-
 /* 80C182DC-80C18340 0000BC 0064+00 1/1 0/0 0/0 .text            __ct__Q29nObjMHasu12daObjMHasu_cFv
  */
-nObjMHasu::daObjMHasu_c::daObjMHasu_c() {
-    // NONMATCHING
+nObjMHasu::daObjMHasu_c::daObjMHasu_c() :
+    mConst1(25.5f),
+    mConst2(6.5f),
+    mConst3(5.5f)
+{
 }
 
 /* 80C18340-80C183D0 000120 0090+00 1/1 0/0 0/0 .text create1st__Q29nObjMHasu12daObjMHasu_cFv */
-void nObjMHasu::daObjMHasu_c::create1st() {
-    // NONMATCHING
+int nObjMHasu::daObjMHasu_c::create1st() {
+    cPhs__Step phase = (cPhs__Step)dComIfG_resLoad(static_cast<request_of_phase_process_class*>(this), l_arcName);
+    if (phase == cPhs_COMPLEATE_e)
+    {
+        setMtx();
+        phase = (cPhs__Step)MoveBGCreate(l_arcName, l_dzbidx, dBgS_MoveBGProc_TypicalRotY, 0x1450, &this->mMtx);
+        if (phase == cPhs_ERROR_e)
+        {
+            return phase;
+        }
+    }
+    return phase;
 }
 
 /* 80C183D0-80C18440 0001B0 0070+00 2/2 0/0 0/0 .text setMtx__Q29nObjMHasu12daObjMHasu_cFv */
 void nObjMHasu::daObjMHasu_c::setMtx() {
-    // NONMATCHING
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    MTXCopy(mDoMtx_stack_c::get(), mMtx2);
+    MTXCopy(mDoMtx_stack_c::get(), mMtx);
 }
 
 /* 80C18440-80C184B0 000220 0070+00 1/0 0/0 0/0 .text CreateHeap__Q29nObjMHasu12daObjMHasu_cFv */
-void nObjMHasu::daObjMHasu_c::CreateHeap() {
-    // NONMATCHING
+int nObjMHasu::daObjMHasu_c::CreateHeap() {
+    J3DModelData* model_data = (J3DModelData*)dComIfG_getObjectRes(l_arcName, l_bmdidx);
+    JUT_ASSERT(0xc1, model_data != 0);
```

Comment:
Defines in assertion strings get converted to their original value due to how nested defines work.
```suggestion
    JUT_ASSERT(0xc1, model_data != NULL);
```

Reply:
Furthermore, please convert the hex val into decimal for these assert/panic messages.